File: python_demo_programs/class_2025_06_21/class3.py
'''
control flow
- conditional
- loop

conditional
- if statement

'''

# number = int(input('Enter a number:'))
# if number > 0:
#     print('positive') # indentation
# elif number < 0:
#     print('negative')
# else:
#     print('zero')


# number = 120
#
# # logical error
# if number > 1:
#     print('number is bigger than 1')
# elif number > 10:
#     print('number is bigger than 10')
# elif number > 100:
#     print('number is bigger than 100')
#
# # fix
# if number > 100:
#     print('number is bigger than 100')
# elif number > 10:
#     print('number is bigger than 10')
# elif number > 1:
#     print('number is bigger than 1')

# number1 = int(input('Enter the first number:'))
# number2 = int(input('Enter the second number:'))
#
# if number1 > number2:
#     print('First number is bigger')
# elif number1 < number2:
#     print('Second number is bigger')
# else:
#     print('Same')

# number = int(input('Enter a number:'))
#
# if number % 2 == 0:
#     print('Even')
# else:
#     print('Odd')

number1 = int(input('Enter the first number:'))
number2 = int(input('Enter the second number:'))
number3 = int(input('Enter the third number:'))

# nested if statement
if number1 > number2:
    if number1 > number3:
        print('First number is the largest')
    else:
        print('Third number is the largest')
else:
    if number2 > number3:
        print('Second number is the largest')
    else:
        print('Third number is the largest')

# option2
if number1 > number2 and number1 > number3:
    print('First number is the largest')
elif number2 > number1 and number2 > number3:
    print('Second number is the largest')
else:
    print('Third number is the largest')

File: cpp_programs/class3.cpp
#include <iostream>
#include <string>

using namespace std;

/*
control flow:
- conditional
    - if statement
    -
- loop

boolean operator
&& and -> one false, overall is false
|| or -> one true, overall is true
! not

true && true -> true
true && false -> false
false && true -> false
false && false -> false

true || true -> true
true || false -> true
false || true -> true
false || false -> false

*/
int main()
{
    // int a = 3;
    // if (a > 0)
    // {
    //     cout << "Positive" << endl;
    // }
    // else if (a < 0)
    // {
    //     cout << "Negative" << endl;
    // }
    // else
    // {
    //     cout << "Zero" << endl;
    // }

    // int grades = 84;
    // if (grades >= 90)
    // {
    //     cout << "A" << endl;
    // }
    // // 80 <= grade < 90
    // // grade < 80 or grade > 90
    // else if (80 <= grades && grades < 90)
    // {
    //     cout << "B" << endl;
    // }
    // else if (70 <= grades && grades < 80)
    // {
    //     cout << "C" << endl;
    // }
    // else
    // {
    //     cout << "D" << endl;
    // }

    int number = 20;

    // logical error
    if (number > 1)
    {
        cout << "Number is bigger than 1" << endl;
    }
    else if (number > 10)
    {
        cout << "Number is bigger than 10" << endl;
    }
    else if (number > 100)
    {
        cout << "Number is bigger than 100" << endl;
    }

    // fix
    if (number > 100)
    {
        cout << "Number is bigger than 100" << endl;
    }
    else if (number > 10)
    {
        cout << "Number is bigger than 10 Number is bigger than 1" << endl;
    }
    else if (number > 1)
    {
        cout << "Number is bigger than 1" << endl;
    }

    // int a, b;
    // cin >> a;
    // cin >> b;

    // if (a > b)
    // {
    // }
    // else if (a < b)
    // {
    // }
    // else
    // {
    // }

    // int a = 3;
    // int b = 4;
    // cout << a % b << endl;

    // if (a % 2 == 0) {

    // }else if (a % 2 != 0) {

    // }

    int a, b, c;
    // find the largest number out of three numbers

    return 0;
}